planning/behavior_path_planner/src/scene_module/lane_change/util.cpp

```diff
@@ -84,14 +86,28 @@ double getDistanceWhenDecelerate(
   return std::max(distance, minimum_distance);
 }
 
+std::pair<double, double> calcLaneChangingSpeedAndDistanceWhenDecelerate(
+  const double velocity, const double shift_length, const double acceleration,
+  const double minimum_lane_change_length, const LaneChangeParameters & params)
+{
+  const auto required_time = PathShifter::calcShiftTimeFromJerkAndJerk(
+    shift_length, params.lane_changing_lateral_jerk, params.lane_changing_lateral_acc);
```

Comment:
@zulfaqar-azmi-t4 minimum_lane_changing_distance is calculated with more complicated logic, so moving it to a common function might be difficult.
My idea is to have only a `minimum_lane_change_distance` in the common param and, the vehicle stops with the margin at the end of lane. Then even if the available distance is shorter than expected, use the shorter one in the lane change module. We can use it as long as the resultant path is not dangerous.

Reply:
Done in https://github.com/autowarefoundation/autoware.universe/pull/2428/commits/6d04c4f1c46a58f0fa02d17e88a14b6ddfb0130b